src/joystick/windows/SDL_windows_gaming_input.c

```diff
@@ -222,7 +222,7 @@ static ULONG STDMETHODCALLTYPE IEventHandler_CRawGameControllerVtbl_Release(__FI
     return 1;
 }
 
-static HRESULT STDMETHODCALLTYPE IEventHandler_CRawGameControllerVtbl_InvokeAdded(__FIEventHandler_1_Windows__CGaming__CInput__CRawGameController * This, IInspectable *sender, __x_ABI_CWindows_CGaming_CInput_CIRawGameController *e)
+static HRESULT STDMETHODCALLTYPE IEventHandler_CRawGameControllerVtbl_InvokeAdded(__FIEventHandler_1_Windows__CGaming__CInput__CRawGameController * This, IInspectable *sender, __x_ABI_CWindows_CGaming_CInput_CIRawGameController **e)
```

Comment:
Sam left the warnings as they are intentionally. See https://github.com/libsdl-org/SDL/commit/dd2f91118e8a44194c21d4cc38ffceb0c7055044 (#4998)

Reply:
the function is used only internally, so how can it break anything?

Reply:
the function is used only internally, so how can it break anything?

Reply:
It's part of a vtable for a COM object, so it can be called outside of SDL's code.

Can we coerce this to remove the compiler warning for older SDKs with an appropriate cast somewhere? We just need to make the compiler not complain; older versions of Windows will still _call_ this function correctly, even if the headers are wrong.

Reply:
If an outside code calls with *parameter, then it is a client/user error.
There is no reference to outside calls in dd2f91118e8a44194c21d4cc38ffceb0c7055044.
It just fixes the internal uses of those functions, but imho the wrong way.
The internal users passed *parameter, while the function was expecting **parameter.
With dd2f91118e8a44194c21d4cc38ffceb0c7055044, the internal users pass *parameter and the functions expect *parameter. In my opinion it should have been fixed the way around. Both the users and the functions should use **parameters.
Can anyone link to a document which says that the correct way is to use *parameters? Google does not help me...

Reply:
Ok, found it at the end of #4998: [sdk patch](https://github.com/hughbe/windows-sdk-headers/blob/e83bf2af3df09ec5420725bd79ccfe1bb534722c/Include/10.0.19041.0/winrt/windows.gaming.input.h#L5554-L5568)
Should this be done with SDK `#ifdefs`?